nsis/src/nsis/format/NsisScriptHeader.java

```diff
@@ -21,8 +21,9 @@
 	static {
 		STRUCTURE = new StructureDataType("script_header", 0);
 		STRUCTURE.add(STRING, NsisConstants.NSIS_MAGIC.length, "magic", null);
-		STRUCTURE.add(DWORD, DWORD.getLength(), "inflated_header_size", null);
 		STRUCTURE.add(DWORD, DWORD.getLength(), "header_size", null);
+		STRUCTURE.add(DWORD, DWORD.getLength(), "length_of_following_data",
```

Comment:
I think everywhere you use the StructureDataType object we don't have a any comments.

I think it makes sense to align all the names with what is in the nsis .h file.

Also, I think that the NSIS_MAGIC actually includes too much data if we want to mirror the structure definted in the .h file.

In this case:
```
typedef struct
{
  int flags; // FH_FLAGS_*
  int siginfo;  // FH_SIG

  int nsinst[3]; // FH_INT1,FH_INT2,FH_INT3

  // these point to the header+sections+entries+stringtable in the datablock
  int length_of_header;

  // this specifies the length of all the data (including the firstheader and CRC)
  int length_of_all_following_data;
} firstheader;
```

For example:
```java
STRUCTURE = new StructureDataType("script_header", 0);
STRUCTURE.add(STRING, NsisConstants.NSIS_SIGINFO, "siginfo", "0xdeadbeef (FH_SIG)
STRUCTURE.add(STRING, NsisConstants.NSIS_MAGIC.length, "nsinst", "NSIS magic (FH_INT1, FH_INT2, FH_INT3)");
STRUCTURE.add(DWORD, DWORD.getLength(), "length_of_header", "Comment");
STRUCTURE.add(DWORD, DWORD.getLength(), "length_of_all_following_data", "Good comment""
STRUCTURE.add(DWORD, DWORD.getLength(), "compressed_header_size", "If MSB is set following data is compressed");
```

Reply:
I like the idea of separating SIGINFO and the magic. When you say we don't have comments where we use the StructureDataType, you mean in the Ghidra interface or in the code?

Reply:
I mean the 4th parameter to the `add()` function, it's for comments. I think it is null everywhere.

Reply:
Nvm I see it in the Struct, all good